shell/shared/grain.js

```diff
@@ -411,7 +411,11 @@ Router.map(function () {
         result.appOrigin = document.location.protocol + "//" + makeWildcardHost(session.hostId);
         setCurrentSessionId(session._id, result.appOrigin, grainId);
         result.sessionId = session._id;
-        result.path = encodeURIComponent("/" + (self.params.path || ""));
+        var currentPath = Iron.Location.get().path;
```

Comment:
I think it makes sense to pull the path and hash from the same place for clarity. In fact if you just start with `window.location` as a string, it'll already include `hash`, right?